### DISCUSSION TITLE
Justify all the text excluding centered paragraphs

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4930] [Q&A] [answered]
To justify all the text excluding the centered parts, I can supplement `align(center)` with `par(justify: false)`:

```
#set page(margin: (x: 0.5in, y: 0.5in), paper: "a5")
#set par(justify: true)

#align(center, par(justify: false)[
  *LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT, SED DO EIUSMOD TEMPOR INCIDIDUNT*
])

#lorem(50)
```

But how to do the same thing using `.where`, so that it won't be required to add `justify: false` again and again to each centered paragraph?

```
#set par.where ....... (justify: true)

#align(center)[
  *LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT, SED DO EIUSMOD TEMPOR INCIDIDUNT*
])

#lorem(50)

#align(center)[
  *LOREM IPSUM DOLOR SIT AMET, CONSECTETUR ADIPISCING ELIT, SED DO EIUSMOD TEMPOR INCIDIDUNT*
])

#lorem(50)
```

**Comment by MDLC01** [comment, marked as answer]
Typically, you would create a new function that encapsulates the desired styling:
```typ
#let important(body) = {
  set align(center)
  set par(justify: false)
  emph(body)
}
```

If you really wanted to use a set rule, you could try something like the following (using a show-set rule), but this does not work in this specific case, likely because `align` is a weird element (not completely sure, though):
```typ
#set par(justify: true)
#show align.where(alignment: center): set par(justify: false)
```

**Reply by johnmapeson** [reply]
Could you elaborate how to use this function?

I tried, for example,

```
#set par(justify: true)

#let important(body) = {
  set align(center)
  set par(justify: false)
  emph(body)
}

#align(center)[
  *Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor xxxxxxxxxxxx ut labore et dolore magna aliqua*
]

#lorem(50)
```

But this doesn't work: all the text is justified. And if I remove `#set par(justify: true)` all the text is flushed left.

**Reply by MDLC01** [reply]
The following should work:

```typ
#set par(justify: true)

#let important(body) = {
  set align(center)
  set par(justify: false)
  emph(body)
}

#important[
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor xxxxxxxxxxxx ut labore et dolore magna aliqua
]

#lorem(50)
```

**Comment by johnmapeson** [comment]
Thank you. The first version works, though maybe someone here could suggest how to use a `set` rule for this. This would be even better solution.

**Comment by xkevio** [comment]
@MDLC01's `show-set`-rule will work with a minor modification: `.where(alignment: center + top)` as by default everything also has a vertical alignment.

**Reply by laurmaedje** [reply]
While this may work, I wouldn't recommend using it. Defining a function is much more robust as this solution will not work with any kind of set-rule-based alignment.

**Comment by hurzl** [comment]
Wouldn't it be expected behaviour for centered text not to be justified by default?

**Reply by laurmaedje** [reply]
Justification and alignment are separate in Typst because even with justification enabled, alignment affects the last line.
